components/openzwave-adapter/src/lib.rs

```diff
@@ -352,6 +385,12 @@ impl OpenzwaveAdapter {
                         if kind.is_none() { continue }
                         let kind = kind.unwrap();
 
+                        let mut tags = HashSet::new();
+                        let label = vid.get_label();
+                        if !label.is_empty() {
+                            tags.insert(TaxoId::<TagId>::new(&vid.get_label()));
```

Comment:
nit: you don't need the `<TagId>` because it's properly infered through the called method's prototype.

And so you can remove it from the import line above.